core/domain/user_domain.py

```diff
@@ -147,7 +202,9 @@ def __init__(
         self.deleted = deleted
         self.created_on = created_on
 
-    def validate(self):
+    # TODO(#15136): Limit the use of type validation. Once all the python files
```

Comment:
I don't think I understand why we need this issue. Even if we finished all the mypy annotations, wouldn't the concern in https://github.com/oppia/oppia/pull/15057#pullrequestreview-907239144 still apply?

Reply:
Maybe we are doing this just to keep track of all the type validations. So that we can remove these type validations wherever possible.

@vojtechjelinek -- it would be great if you suggest something on this because I added these changes by obeying both comments!

Reply:
All schema handlers are going to be covered under "core/controllers". So, I removed the todo.
Done.

Reply:
@sahiljoster32 You haven't closed or updated the issue, so I don't think this comment thread can be considered resolved yet. 

Just as a check, could you perhaps reply to this thread with an explanation of your understanding of what the concern is? I think this would help avoid miscommunications.

Reply:
Yeah sure.
MyPy does not check typing at runtime. So, if we remove the type validation then there might be a chance that we get data of inappropriate type through controllers which can break things and leads to untrackable errors. So, to avoid that scenario we should keep these type validations.

Why we are not adding TODO comments? reason below:-
The schema handlers that are responsible for passing the data are going to be type annotated once all the domain files are covered (type annotated). 

Also, I closed issue #15136.
I hope this sounds good!!

Reply:
This seems fine, just note that, when closing an issue or a PR, always leave a comment with the reason. (Please do that for #15136.)

For your summary, just a note -- the only thing that will prevent runtime errors is the handler schema validation, not the mypy type-annotation. You sort of mix these two things up in the second part of your comment, so I wanted to clarify that.

Reply:
Thank you so much for clarification. I'll keep that in mind.